pkg/inspector/inspector.go

```diff
@@ -114,17 +118,20 @@ func (i *Inspector) Send(ctx context.Context, r record.Record) {
 	}
 }
 
-func (i *Inspector) NewSession(ctx context.Context) *Session {
+func (i *Inspector) NewSession(ctx context.Context, componentID string) *Session {
```

Comment:
As I was reviewing the PR myself, I realized it's better to have this in the inspector. After all, a single inspector manages sessions for only one component (connector or processor).

Reply:
... or no. Processors cannot be easily refactored. 

Processors themselves (i.e. `processor.Interface`) are created through builders. Builders (currently) only accept a processor configuration. Changing the builder would mean we'd change many places in the code.

Another approach is to create an inspector in `processor.Service.Create()`. A decorator would add inspection to all processors. `processor.Service` has all the important information needed for the metrics.

That would have a nice side-effect of simplifying the `procbuiltin.FuncWrapper` and `procjs.Processor` (it would reduce them to just one method). However, `processor.Store` is also building processors, so we'd need to duplicate the logic there as well.

All in all, I believe this is doable, but I think it would unnecessarily increase the scope of the PR. I'll try to refactor this in a separate one.

Reply:
I'd say that if we pour any amount of work into refactoring inspectors, it should be to move them to nodes and add an inspector wrapper node (or a normal inspector node that's added between other nodes). Then the logic would be contained to a single place, while connectors and processors wouldn't even know there is such a thing as an inspector. But I don't think this needs to be refactored right now, it's good enough and it works 👍